src/main/java/subway/models/Sections.java

```diff
@@ -31,41 +37,38 @@ public List<Station> getStations() {
     }
 
     public void validateUpStationForAdd(Station upStation) {
-        if (!isDownwardEndPoint(upStation)) {
-            throw new IllegalArgumentException();
+        if (isNotDownwardEndPoint(upStation)) {
+            throw new IllegalArgumentException("상행역은 노선의 하행종점이어야 합니다.");
         }
     }
 
     public void validateDownStationForAdd(Station downStation) {
         if (containsStation(downStation)) {
-            throw new IllegalArgumentException();
+            throw new IllegalArgumentException("하행역은 노선에 등록된 역일 수 없습니다.");
         }
     }
 
     public void validateStationForRemove(Station station) {
         if (hasLessThenTwoSections()) {
-            throw new IllegalArgumentException();
+            throw new IllegalArgumentException("노선에 두개 이상의 구간이 있어야 삭제가 가능합니다.");
         }
-        if (!isDownwardEndPoint(station)) {
-            throw new IllegalArgumentException();
+        if (isNotDownwardEndPoint(station)) {
+            throw new IllegalArgumentException("노선의 하행종점역만 삭제가 가능합니다.");
         }
     }
 
-    private boolean isDownwardEndPoint(Station station) {
-        Station downwardEndPoint = sections.get(sections.size() - 1).getDownStation();
-        return downwardEndPoint.getId().equals(station.getId());
+    private boolean isNotDownwardEndPoint(Station station) {
+        int lastIndex = sections.size() - 1;
+        Station downwardEndPoint = sections.get(lastIndex).getDownStation();
+        return !downwardEndPoint.equals(station);
     }
 
     private boolean containsStation(Station station) {
-        for (Station s : getStations()) {
-            if (station.getId().equals(s.getId())) {
-                return true;
-            }
-        }
-        return false;
+        return getStations().stream().anyMatch(s -> s.equals(station));
```

Comment:
```suggestion
        return getStations().contains(station);
```

리팩토링 💯 
이렇게 할수도 있을것 같아요 👍